### PR TITLE
classnames: let classNames() to accept boolean, as the original implementation does

### PR DESCRIPTION
**Pull request by gfx**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

---

ref. https://github.com/JedWatson/classnames/blob/master/index.js#L22

It just ignores an arg unless the arg is not string, number, array, nor object. That is, the original impl just ignores boolean.

### CODE REVIEW

File: types/classnames/classnames-tests.ts [hunk 1]
```diff
@@ -38,3 +38,6 @@ const className = cx('foo', ['bar'], { baz: true }); // => "abc def xyz"
 
 // falsey values are just ignored
 cx(null, 'bar', undefined, 0, 1, { baz: null }, ''); // => 'bar 1'
+
+// true is just ignored
+cx(true || "foo");
```
Comment (JKillian) [inline types/classnames/classnames-tests.ts]: what's the point of `|| "foo"` here?

**Comment by JKillian** [comment]
Technically the docs for classnames just say:

> The classNames function takes any number of arguments which can be a string or object.

I'm a little hesitant to just change the type info based on an implementation detail, in general, it seems like it would be an error to accidentally be passing `true` to `classNames`. What's your usecase here @gfx ?

File: types/classnames/classnames-tests.ts (hunk 1 above)
Comment (gfx) [reply types/classnames/classnames-tests.ts]: This is a typical use case in my code.;`classNames(true)` is useless, but `classNames(constExprToBeTrue || className)` is the case what I want to use.

**Comment by gfx** [comment]
@JKillian Thanks for your review.

As I commented at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25755#discussion_r188177203, `className(booleanExpr || className)` or `className(booleanExpr && className)` is the use cases. They are the same as `className({ className: booleanExpr })` but `className(booleanExpr || className)` is fastter in theory because it does not create a new object.

In addition, it is in the test: https://github.com/JedWatson/classnames/blob/master/tests/index.js#L47-L49

**Comment by DanielRosenwasser** [comment]
> It just ignores an arg unless the arg is not string, number, array, nor object. That is, the original impl just ignores boolean.

Seems like you really just want `any` at that point, don't you?

**Comment by gfx** [comment]
@DanielRosenwasser 

I never said I want `any`. Rather, I just want  to use boolean.

The original `classnames`' [usage](https://github.com/JedWatson/classnames) suggests `false` is a kind of use-case, which then suggests boolean expressions, I guess. Whoever uses a constant `false` for `classNames()`? A boolean expression is sometimes false, but sometimes true.

**Comment by DanielRosenwasser** [comment]
I think given that it's in the test suite, this is reasonable.

**Comment by gfx** [comment]
Thanks!

**Pull request merged**